active-rfcs/0000-function-api.md

```diff
@@ -136,7 +136,49 @@ const MyComponent = {
 
 Note this `props` object is reactive - i.e. it is updated when new props are passed in, and can be observed and reacted upon using the `watch` function introduced later in this RFC. However, for userland code, it is immutable during development (will emit warning if user code attempts to mutate it).
 
-`this` is usable inside `setup()`, but you most likely won't need it very often.
+The second argument provides a context object which exposes a number of properties that were previously exposed on `this` in 2.x APIs:
```

Comment:
What would be our strategy to deal with custom instance properties, i.e. `Vue.prototype.$something` added by a plugin?

Reply:
Plugins written for the new API will likely be providing functions like `useXXX` to directly provide reactive values into a component.

For legacy plugins, we can make `this` available, but in the compatibility build only.

Reply:
> 
> 
> Plugins written for the new API will likely be providing functions like `useXXX` to directly provide reactive values into a component.
> 
> For legacy plugins, we can make `this` available, but in the compatibility build only.

I am mistaken in the belief that you are massively underestimating how many popular plugins, that use/need the "this" keyword, will never be written. Many great/popular plugins that the community currently uses have zero updates/commits for 18+ months, and seem pretty much abandoned.